### PR TITLE
Cluster-autoscaler - check generic predicates

### PR DESCRIPTION
**Pull request by mwielgus**
kubernetes/kubernetes#24404

Requires that https://github.com/kubernetes/contrib/pull/866 is merged before.

cc: @piosz @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/simulator/predicates.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+// NewPredicateChecker builds PredicateChecker.
+func NewPredicateChecker() *PredicateChecker {
+	// TODO(fgrzadkowsi): Get a full list of all predicates.
+	return &PredicateChecker{}
+}
+
+// CheckPredicates Checks if the given pod can be placed on the given node.
+func (p *PredicateChecker) CheckPredicates(pod *kube_api.Pod, node *kube_api.Node, nodeInfo *schedulercache.NodeInfo) error {
+
```
Comment (piosz) [inline cluster-autoscaler/simulator/predicates.go]: empty line

File: cluster-autoscaler/utils.go [hunk 2]
```diff
@@ -145,17 +145,18 @@ func CheckMigsAndNodes(nodes []*kube_api.Node, gceManager *gce.GceManager) error
 }
 
 // GetNodeInfosForMigs finds NodeInfos for all migs used to manage the given nodes.
-func GetNodeInfosForMigs(nodes []*kube_api.Node, gceManager *gce.GceManager, kubeClient *kube_client.Client) (map[string]*schedulercache.NodeInfo, error) {
+func GetNodeInfosForMigs(nodes []*kube_api.Node, gceManager *gce.GceManager, kubeClient *kube_client.Client) (map[string]*schedulercache.NodeInfo,
```
Comment (piosz) [inline cluster-autoscaler/utils.go]: What is the result of this function?

File: cluster-autoscaler/simulator/predicates.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/simulator/predicates.go]: Removed.

**Comment by piosz** [comment]
LGTM, please squash commits to have 2: one with godeps, second one with the real changes.

File: cluster-autoscaler/utils.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/utils.go]: Done.

File: cluster-autoscaler/simulator/predicates.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/simulator/predicates.go]: Removed.

**Pull request merged**